apps/web/app/api/slack/callback/route.ts

```diff
@@ -0,0 +1,67 @@
+import { responses } from "@/app/lib/api/response";
+import { NextRequest } from "next/server";
+
+import { SLACK_CLIENT_ID, SLACK_CLIENT_SECRET, WEBAPP_URL } from "@formbricks/lib/constants";
+import { createOrUpdateIntegration } from "@formbricks/lib/integration/service";
+import { TIntegrationSlackConfig, TIntegrationSlackCredential } from "@formbricks/types/integration/slack";
+
+export async function GET(req: NextRequest) {
+  const url = req.url;
+  const queryParams = new URLSearchParams(url.split("?")[1]); // Split the URL and get the query parameters
+  const environmentId = queryParams.get("state"); // Get the value of the 'state' parameter
+  const code = queryParams.get("code");
+  const error = queryParams.get("error");
+
+  if (!environmentId) {
+    return responses.badRequestResponse("Invalid environmentId");
+  }
+
+  if (code && typeof code !== "string") {
+    return responses.badRequestResponse("`code` must be a string");
+  }
+
+  const client_id = SLACK_CLIENT_ID;
```

Comment:
Is this step necessary? We could also continue with `SLACK_CLIENT_ID`.
(Also otherwise a camel-case `clientId` would be the right choice)